filters/openpolicyagent/openpolicyagent.go

```diff
@@ -205,7 +208,31 @@ func WithStartupTimeout(timeout time.Duration) func(*OpenPolicyAgentInstanceConf
 }
 
 func (cfg *OpenPolicyAgentInstanceConfig) GetEnvoyMetadata() *ext_authz_v3_core.Metadata {
-	return cfg.envoyMetadata
+	if cfg.envoyMetadata == nil {
+		return nil
+	} else {
+		return DeepCopy(cfg.envoyMetadata)
```

Comment:
I think there is a standard method to clone protobuf message https://pkg.go.dev/github.com/golang/protobuf/proto#Clone:

```go
func (cfg *OpenPolicyAgentInstanceConfig) GetEnvoyMetadata() *ext_authz_v3_core.Metadata {
	if cfg.envoyMetadata != nil {
		return proto.Clone(cfg.envoyMetadata).(*ext_authz_v3_core.Metadata)
	}
	return nil
}
```
Could you test if that works?